src/CMakeLists.txt

```diff
@@ -84,10 +97,17 @@ elseif(MINGW)
 
 endif()
 
-# Set the MSVC debug working directory, enforce conformance mode for osp-magnum
+# MSVC quality of life improvements for Visual Studio
 if(MSVC)
-  set_property(TARGET osp-magnum PROPERTY VS_DEBUGGER_WORKING_DIRECTORY ../../bin)
-  target_compile_options(osp-magnum PRIVATE $<$<CXX_COMPILER_ID:MSVC>:/permissive->)
+	# Set the MSVC debug working directory
+  	set_property(TARGET osp-magnum PROPERTY VS_DEBUGGER_WORKING_DIRECTORY ../../bin)
+
+  	# Enforce conformance mode for osp-magnum
+  	target_compile_options(osp-magnum PRIVATE $<$<CXX_COMPILER_ID:MSVC>:/permissive->)
+
+  	# Segregate headers, shaders into filters
+  	source_group("Shader Files" FILES ${SHADER_FILES})
+  	source_group("Header Files" FILES ${H_FILES})
```

Comment:
Can you also add `set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT osp-magnum)`
to this? This makes it so you don't have to change the startup project, which makes it a bit more convenient.

Reply:
Yep, on it